virttest/libvirt_vm.py

```diff
@@ -2253,7 +2254,10 @@ def destroy(self, gracefully=True, free_mac_addresses=True):
 
     def remove(self):
         self.destroy(gracefully=True, free_mac_addresses=False)
-        if not self.undefine():
+        undef_opts = "--managed-save --snapshots-metadata"
+        if libvirt_version.version_compare(7, 5, 0):
+            undef_opts += " --checkpoints-metadata"
```

Comment:
@fangge1212 , could you help explain the reason we **always** have to use '--managed-save --snapshots-metadata' and '--checkpoints-metadata' after libvirt-7.5.0 when undefining a vm?

Reply:
If vm has managedsave file or snapshots or checkpoints, it will affect vm undefine, e.g.
# virsh undefine vm1
error: Failed to undefine domain 'vm1'
error: Requested operation is not valid: cannot delete inactive domain with 8 checkpoints

And there is no side effect by adding the options if vm has no managedsave file/snapshots/checkpoints

Reply:
I saw the problems you faced but I intend to unfix the options. It's a common function, someone may need to catch the errors like above. I think you can make it configurable or define your own function to force remove a domain. Or bring this topic to the pr review meeting to discuss with others.